src/projectworks.ts

```diff
@@ -0,0 +1,83 @@
+import fetch, { Headers, Response } from 'node-fetch';
+import { URLSearchParams } from 'url';
+import type { ProjectWorksLeave, ProjectWorksUser } from './types';
+
+const PW_URL = 'https://api.projectworksapp.com';
+const PW_USERNAME = process.env.PROJECTWORKS_USERNAME;
+const PW_PASSWORD = process.env.PROJECTWORKS_PASSWORD;
+
+const isObjectWithPropName = (ob: unknown, propName: string): boolean => {
+  if (!ob) {
+    return false;
+  }
+
+  if (typeof ob !== 'object') {
+    return false;
+  }
+
+  if (!(propName in ob)) {
+    return false;
+  }
+
+  return true;
+};
```

Comment:
I'm tempted to replace this with `lodash` tbh. I know lodash in client-side code is tricky because it can be a big dependency but that wouldn't be an issue here. 

Built-in JS does quite a lot but there are gaps (like this) - how sad would you be if we pulled in lodash @G-Rath ?

Reply:
I was actually thankful you hadn't brought in `lodash` for this 😅

`lodash` is still a very big dependency which not really maintained anymore, and you'd be pulling it in to cover two usages of a small function you've already written, which really you've got because we've not properly typed out the PW API responses - so I'd rather we did that then bring in `lodash`

Reply:
I have pulled these functions into a module and mimicked the lodash API because I think it's quite clean and composable for these kinds of tests and because I want a file I can easily paste into other apps (I seem to hit a "wishing I had lodash" point on every lambda I work on)